### PR TITLE
Fix (Legacy Maps): Add necessary specificity for dark mode style over…

### PR DESCRIPTION
**Pull request by joshuarrrr**
…rides

### Description

- wrap override styles in visualization selector
- remove temp SASS var
- update attribution background to be opaque for consistency with other controls
- add raster tile filter to map tiles in dark mode, since OpenSearch doesn't serve dark mode raster tilesets
- Fix tooltip behavior so that tooltip only appears when there's content to render

### Issues Resolved

fixes https://github.com/opensearch-project/dashboards-maps/issues/449, fixes https://github.com/opensearch-project/dashboards-maps/issues/450

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/1679762/ab67cc26-7cfd-4b57-8362-b0b519be029b

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/maps_legacy/public/map/_leaflet_overrides.scss [hunk 1]
```diff
@@ -1,164 +1,166 @@
...
+    padding: 0;
+    margin: 0;
 
-  font-family: $euiFontFamily;
-  font-weight: $euiFontWeightMedium;
-  line-height: $euiLineHeight;
+    @include fontSize(11px);
 
-  label {
+    font-family: $euiFontFamily;
```
Comment (AMoo-Miki) [inline src/plugins/maps_legacy/public/map/_leaflet_overrides.scss:71]:
```suggestion
    font-family: var(--font-text);
```

**Review by AMoo-Miki — APPROVED** [review]

**Review by KrooshalUX — APPROVED** [review]
From a UX perspective, LGTM!

File: src/plugins/maps_legacy/public/map/_leaflet_overrides.scss (hunk 1 above)
Comment (joshuarrrr) [reply src/plugins/maps_legacy/public/map/_leaflet_overrides.scss:71]: I was going to defer this to a separate issue and PR, but I realized that this plugin is the only one left in core using `$euiFontFamily` directly, so I'll change here. But I think we should actually add a stylelint rule to enforce this if it's important.

Comment (joshuarrrr) [reply src/plugins/maps_legacy/public/map/_leaflet_overrides.scss]: fixed

Comment (joshuarrrr) [reply src/plugins/maps_legacy/public/map/_leaflet_overrides.scss]: issue created: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4660

**Review by AMoo-Miki — APPROVED** [review]

File: src/plugins/maps_legacy/public/map/_leaflet_overrides.scss [hunk 2]
```diff
@@ -1,164 +1,166 @@
...
 
-  @include fontSize(11px);
+  .leaflet-control-layers-expanded {
+    padding: 0;
+    margin: 0;
 
-  font-family: $euiFontFamily;
-  font-weight: $euiFontWeightMedium;
-  line-height: $euiLineHeight;
+    @include fontSize(11px);
```
Comment (BSFishy) [inline src/plugins/maps_legacy/public/map/_leaflet_overrides.scss:69]: Would love to use some variant of `euiSize` here, but I think that's outside the scope of this PR

**Review by BSFishy — APPROVED** [review]

File: src/plugins/maps_legacy/public/map/_leaflet_overrides.scss (hunk 2 above)
Comment (joshuarrrr) [reply src/plugins/maps_legacy/public/map/_leaflet_overrides.scss:69]: Yeah, agreed. I imagine we'll tackle such things as part of https://github.com/opensearch-project/oui/issues/817

**Pull request merged**